js/mock_runtime.js

```diff
@@ -41,22 +37,20 @@ global.SendByteLength = () => {
 };
 
 global.RecvReturnEmpty = () => {
-  const ui8 = new Uint8Array("abc".split("").map(c => c.charCodeAt(0)));
-  const ab = typedArrayToArrayBuffer(ui8);
-  let r = deno.send(ab);
-  assert(r == null);
-  r = deno.send(ab);
-  assert(r == null);
+  const m1 = new Uint8Array("abc".split("").map(c => c.charCodeAt(0)));
+  const m2 = m1.slice();
+  const r1 = deno.send(m1);
+  assert(r1 == null);
+  const r2 = deno.send(m2);
+  assert(r2 == null);
 };
 
 global.RecvReturnBar = () => {
-  const ui8 = new Uint8Array("abc".split("").map(c => c.charCodeAt(0)));
-  const ab = typedArrayToArrayBuffer(ui8);
-  const r = deno.send(ab);
-  assert(r instanceof ArrayBuffer);
+  const m = new Uint8Array("abc".split("").map(c => c.charCodeAt(0)));
+  const r = deno.send(m);
+  assert(r instanceof Uint8Array);
   assert(r.byteLength === 3);
-  const rui8 = new Uint8Array(r);
-  const rstr = String.fromCharCode(...rui8);
+  const rstr = String.fromCharCode(...r);
```

Comment:
You can expand a Uin8Array as splat arguments...? Odd, but ok, I guess it does Array.from() somewhere....

Reply:
Yes, and you wrote that yourself:

```diff
-  const rstr = String.fromCharCode(...rui8);  // <-- what was there before
+  const rstr = String.fromCharCode(...r);     // <-- I just did s/rui8/r/
```

Reply:
> Odd, but ok, I guess it does Array.from() somewhere....

That's actually not how it works ... no array is being created, Array.from is not involved.
Splatting works by iterating over an iterable object.

```js
let iter = {
  [Symbol.iterator]() { 
    let value = 0, end = 9;
    return { next: () => ({ value, done: value++ > end }) };
  }
};

let arraylike = {0:'a', 1:'b', 2:'c', length:3};

console.log(...iter);       // Works.
console.log(...arraylike);  // Error: arraylike not iterable.
console.log(..."abcde");    // a b c d e -- strings are iterable!

// Array.from works the same way, it expects an iterable.
// But it also has a fallback for non-iterable array-like objects.
console.log(...iter);                   // OK.
console.log(...Array.from(arraylike));  // Also OK.
```